clients/src/main/java/org/apache/kafka/common/internals/PartitionStates.java

```diff
@@ -43,11 +43,11 @@
  */
 public class PartitionStates<S> {
 
-    private final LinkedHashMap<TopicPartition, S> map = new LinkedHashMap<>();
+    private final Map<TopicPartition, S> map = new LinkedHashMap<>();
     private final Set<TopicPartition> partitionSetView = Collections.unmodifiableSet(map.keySet());
 
     /* the number of partitions that are currently assigned available in a thread safe manner */
-    private volatile int size = 0;
+    private int size = 0;
```

Comment:
Any particular reason for this change? We should update the javadoc if we think it was not necessary to make `size()` safe for concurrent access.

Reply:
I don't think it was intentional from me to remove this. Will undo it.